src/v/cluster/producer_state.cc

```diff
@@ -491,12 +502,16 @@ producer_state::snapshot(kafka::offset log_start_offset) const {
     snapshot.ms_since_last_update = ms_since_last_update();
     snapshot.finished_requests.reserve(_requests._finished_requests.size());
     for (auto& req : _requests._finished_requests) {
-        vassert(
-          req->has_completed(),
-          "_finished_requests has unresolved promise: {}, range:[{}, {}]",
-          *this,
-          req->_first_sequence,
-          req->_last_sequence);
+        if (!req->has_completed()) {
+            vlog(
+              _logger.error,
+              "[{}] Ignoring unresolved finished request during snapshot, "
+              "range: "
+              "range:[{}, {}]",
+              *this,
+              *req);
```

Comment:
@bharathv this looks like it has 3 `{}` but 2 parameters?

Reply:
:'(  :hawk - eye: will do a follow up PR.